test/Test/Toml/Codec/Combinator/Custom.hs

```diff
@@ -25,3 +26,6 @@ customSpec = describe "Combinator.Custom: Roundtrip tests" $ do
     codecRoundtrip "TextBy          "
         (Toml.textBy (Text.pack . show) (first Text.pack . readEither . Text.unpack))
         Gen.genInt
+    codecRoundtrip "either          "
+        (Toml.either . Toml.bool)
+        (Right <$> Gen.genBool)
```

Comment:
Here you are testing the `either` combinator only on values generated with the constructor `Right`, but we actually want to test on both `Left` and `Right`. So, we need to generate either a list of decode errors or a different value (in your case it's `Bool`)

Reply:
Hmmm I'm no longer sure if `TomlCodec (Either [TomlDecodeError] a)` is correct :thinking: 

We probably don't want to enode a `Left [TomlDecodeError]`... Maybe we should change `either` to:

```
either :: Codec i o -> Codec i (Either [TomlDecodeError] o)
```

and then we need to use something other than `codecRoundtrip` in the test :thinking: 
```

Reply:
@markus1189 I think it is okay to have `TomlCodec` in type as `TomlDecodeError` works with `TomlCodec` 🙂 
You can just change the generator to something with [`Gen.either`](https://hackage.haskell.org/package/hedgehog-1.0.3/docs/Hedgehog-Gen.html#v:either).